# 発展問題
- 繰り返し回数を増やして比較
- マジックコマンド `%timeit`, `%%timeit` を利用して計算時間を計測

 → `%%timeit` を利用し、セルごとの処理を繰り返し実行・比較する
 
 
なお、変数の定義や `print` 分など共通の処理については除く。

In [1]:
THICKNESS = 8e-5

In [2]:
%%timeit
folded_thickness_1 = THICKNESS * (2**43)

134 ns ± 7.98 ns per loop (mean ± std. dev. of 7 runs, 10000000 loops each)


In [3]:
%%timeit
folded_thickness_2 = THICKNESS
for i in range(43):
    folded_thickness_2 *= 2

3.49 µs ± 357 ns per loop (mean ± std. dev. of 7 runs, 100000 loops each)


# 結果
## べき乗
134 ± 7.98 (ns/loop) : 10^7loop
## for文
> 3.49 µs ± 357 ns per loop

= 3490 ± 357 (ns/loop) : 10^5loop
## 比較
 **for文計算時間 > べき乗計算時間** という結果になった。

内部ロジックの計算量の差が顕著に表れていると考えられる。（詳細については割愛）